### PR TITLE
Working with multiple trackers

### PR DESCRIPTION
**Pull request by mhmoudgmal**

### CODE REVIEW

**Comment by yangshun** [comment]
I'll review this asap. Sorry for the wait.

File: src/index.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ var log = require('./utils/console/log');
 
 var _debug = false;
 var _titleCase = true;
+var _trackerName = '';
```
Comment (yangshun) [inline src/index.js]: `null` would be a better option.

File: src/index.js [hunk 2]
```diff
@@ -118,7 +123,7 @@ var ReactGA = {
    */
   send: function (fieldObject) {
     if (typeof ga === 'function') {
-      ga('send', fieldObject);
+      ga((_trackerName == '' ? '' : _trackerName + '.') + 'send', fieldObject);
```
Comment (yangshun) [inline src/index.js]:
```
(_trackerName == '' ? '' : _trackerName + '.')
```
can be simplified to 

```
(_trackerName && _trackerName + '.')
```

File: src/index.js [hunk 3]
```diff
@@ -145,7 +150,7 @@ var ReactGA = {
     }
 
     if (typeof ga === 'function') {
-      ga('send', 'pageview', path);
+      ga((_trackerName == '' ? '' : _trackerName + '.') + 'send', 'pageview', path);
```
Comment (yangshun) [inline src/index.js]: Probably can extract this appending logic into a function since it's used in multiple places.

**Comment by yangshun** [comment]
Thanks for the PR. I just read up on GA trackers and there are a few outstanding issues with this PR:
- Although this PR aims to support multiple trackers, only one global (within the library scope) `_trackerName` variable is present to keep track of the last created tracker, so this PR doesn't really do what it says.
- Based on GA's [Command Queue Reference](https://developers.google.com/analytics/devguides/collection/analyticsjs/command-queue-reference), `trackerName`s can be used in front of any of the commands, so the tracker name appending logic has to be added to all the library's internal `ga(...)` calling commands. This part probably needs some refactoring as it won't be elegant to manually add this logic to all places we're calling `ga(...)` in the library.
- Lastly, tests have to be added :joy:
 
That being said, it is certainly quite a non-trivial effort to support multiple trackers. Unfortunately, we have to do it right or not do it at all.

**Comment by mhmoudgmal** [comment]
Thanks for your comments. and sorry it was my fault, the commit message was misleading as I was solving the problem where the library was not functioning when it is being used in a widget that was included in another site which was already using google analytics, so I wanted to have the possibility to pass the tracker name in react-ga.

**Comment by yangshun** [comment]
No problem at all! Actually for your use case, you can access the native `ga` function via `ReactGA.ga` and do `ReactGA.ga('clientTracker.send', ...)` to serve your need. There's no real need to modify the library code to support multiple trackers (:

**Pull request closed without merging**